taichi/ir/type.cpp

```diff
@@ -103,6 +103,10 @@ std::string TensorType::to_string() const {
   return s;
 }
 
+size_t TensorType::get_element_offset(int ind) const {
+  return data_type_size(element_) * ind;
```

Comment:
I'm a bit confused about why only the element offset of `TensorType` needs the `data_type_size()` function.

Reply:
We calculate the offset of a `StructType` by calling `llvm::DataLayout::getStructLayout` because we don't know how it pads the struct type. For a nested struct type, we calculate it by adding the struct/tensor offsets of every depth. We don't need `data_type_size()` for struct type because the offsets of scalars are calculated by LLVM.